### ISSUE TITLE
Errrors when checking config file

### ISSUE THREAD

**Issue by litfoo** [robmarkcole/TrasportAPI-HASS#13]
I placed the uk_transport.py file in /custom_components/sensor folder and get the following error when checking my config file before restarting HASS:
```

`2017-07-09 11:22:03 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/lib/python3.4/asyncio/tasks.py", line 233, in _step
    result = coro.throw(exc)
  File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/setup.py", line 61, in async_setup_component
    return (yield from task)
  File "/usr/lib/python3.4/asyncio/futures.py", line 388, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.4/asyncio/tasks.py", line 286, in _wakeup
    value = future.result()
  File "/usr/lib/python3.4/asyncio/futures.py", line 277, in result
    raise self._exception
  File "/usr/lib/python3.4/asyncio/tasks.py", line 237, in _step
    result = next(coro)
  File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/setup.py", line 156, in _async_setup_component
    conf_util.async_process_component_config(hass, config, domain)
  File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/config.py", line 589, in async_process_component_config
    platform = get_platform(domain, p_name)
  File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/loader.py", line 103, in get_platform
    return get_component(PLATFORM_FORMAT.format(domain, platform))
  File "/usr/lib/python3.4/unittest/mock.py", line 896, in __call__
    return _mock_self._mock_call(*args, **kwargs)
  File "/usr/lib/python3.4/unittest/mock.py", line 962, in _mock_call
    ret_val = effect(*args, **kwargs)
  File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/scripts/check_config.py", line 179, in mock_get
    module = MOCKS['get'][1](comp_name)
  File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/loader.py", line 141, in get_component
    module = importlib.import_module(path)
  File "/usr/lib/python3.4/importlib/__init__.py", line 109, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
  File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
  File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1129, in _exec
  File "<frozen importlib._bootstrap>", line 1467, in exec_module
  File "<frozen importlib._bootstrap>", line 1572, in get_code
  File "<frozen importlib._bootstrap>", line 1532, in source_to_code
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "/home/homeassistant/.homeassistant/custom_components/sensor/uk_transport.py", line 7
    <!DOCTYPE html>
    ^
SyntaxError: invalid syntax
(homeassistant) homeassistant@hassbian:/home/pi $
(homeassistant) homeassistant@hassbian:/home/pi $ 2017-07-09 11:22:03 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
bash: syntax error near unexpected token `('
(homeassistant) homeassistant@hassbian:/home/pi $ Traceback (most recent call last):
bash: syntax error near unexpected token `most'
(homeassistant) homeassistant@hassbian:/home/pi $   File "/usr/lib/python3.4/asyncio/tasks.py", line 233, in _step
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     result = coro.throw(exc)
bash: syntax error near unexpected token `('
(homeassistant) homeassistant@hassbian:/home/pi $   File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/setup.py", line 61, in async_setup_component
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     return (yield from task)
bash: syntax error near unexpected token `yield'
(homeassistant) homeassistant@hassbian:/home/pi $   File "/usr/lib/python3.4/asyncio/futures.py", line 388, in __iter__
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     yield self  # This tells Task to wait for completion.
bash: yield: command not found
(homeassistant) homeassistant@hassbian:/home/pi $   File "/usr/lib/python3.4/asyncio/tasks.py", line 286, in _wakeup
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     value = future.result()
bash: syntax error near unexpected token `('
(homeassistant) homeassistant@hassbian:/home/pi $   File "/usr/lib/python3.4/asyncio/futures.py", line 277, in result
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     raise self._exception
bash: raise: command not found
(homeassistant) homeassistant@hassbian:/home/pi $   File "/usr/lib/python3.4/asyncio/tasks.py", line 237, in _step
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     result = next(coro)
bash: syntax error near unexpected token `('
(homeassistant) homeassistant@hassbian:/home/pi $   File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/setup.py", line 156, in _async_setup_component
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     conf_util.async_process_component_config(hass, config, domain)
bash: syntax error near unexpected token `hass,'
(homeassistant) homeassistant@hassbian:/home/pi $   File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/config.py", line 589, in async_process_component_config
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     platform = get_platform(domain, p_name)
bash: syntax error near unexpected token `('
(homeassistant) homeassistant@hassbian:/home/pi $   File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/loader.py", line 103, in get_platform
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     return get_component(PLATFORM_FORMAT.format(domain, platform))
bash: syntax error near unexpected token `('
(homeassistant) homeassistant@hassbian:/home/pi $   File "/usr/lib/python3.4/unittest/mock.py", line 896, in __call__
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     return _mock_self._mock_call(*args, **kwargs)
bash: syntax error near unexpected token `('
(homeassistant) homeassistant@hassbian:/home/pi $   File "/usr/lib/python3.4/unittest/mock.py", line 962, in _mock_call
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     ret_val = effect(*args, **kwargs)
bash: syntax error near unexpected token `('
(homeassistant) homeassistant@hassbian:/home/pi $   File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/scripts/check_config.py", line 179, in mock_get
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     module = MOCKS['get'][1](comp_name)
bash: syntax error near unexpected token `('
(homeassistant) homeassistant@hassbian:/home/pi $   File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/loader.py", line 141, in get_component
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     module = importlib.import_module(path)
bash: syntax error near unexpected token `('
(homeassistant) homeassistant@hassbian:/home/pi $   File "/usr/lib/python3.4/importlib/__init__.py", line 109, in import_module
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     return _bootstrap._gcd_import(name[level:], package, level)
bash: syntax error near unexpected token `('
(homeassistant) homeassistant@hassbian:/home/pi $   File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $   File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $   File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $   File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $   File "<frozen importlib._bootstrap>", line 1129, in _exec
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $   File "<frozen importlib._bootstrap>", line 1467, in exec_module
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $   File "<frozen importlib._bootstrap>", line 1572, in get_code
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $   File "<frozen importlib._bootstrap>", line 1532, in source_to_code
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $   File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $   File "/home/homeassistant/.homeassistant/custom_components/sensor/uk_transport.py", line 7
bash: File: command not found
(homeassistant) homeassistant@hassbian:/home/pi $     <!DOCTYPE html>
bash: !DOCTYPE: event not found
(homeassistant) homeassistant@hassbian:/home/pi $     ^
bash: ^: command not found
(homeassistant) homeassistant@hassbian:/home/pi $ SyntaxError: invalid syntax
bash: SyntaxError:: command not found
`
```

**Comment by robmarkcole** [comment]
Whats in your config (anonymise your api key)?

**Comment by litfoo** [comment]
`  - platform: uk_transport
    app_id: !secret uk_transport_app_id
    app_key: !secret uk_transport_app_key
    live_bus_time:
        - stop_atcocode: 340000368SHE
          direction: Parkway Station
    live_train_time:
        - station_code: WIM
          calling_at: WAT

  - platform: template
    sensors:
      next_train_status:
        friendly_name: 'Next train status'
        value_template: '{{states.sensor.next_train_to_wat.attributes.next_trains[0].status}}'
      next_trains_origin:
        friendly_name: 'Next train origin'
        value_template: '{{states.sensor.next_train_to_wat.attributes.next_trains[0].origin_name}}'
      next_trains_estimated:
        friendly_name: 'Next train estimated'
        value_template: '{{states.sensor.next_train_to_wat.attributes.next_trains[0].estimated}}'
      next_trains_scheduled:
        friendly_name: 'Next train scheduled'
        value_template: '{{states.sensor.next_train_to_wat.attributes.next_trains[0].scheduled}}'
      next_trains_platform:
        friendly_name: 'Next train platform'
        value_template: '{{states.sensor.next_train_to_wat.attributes.next_trains[0].platform}}'
   
`

**Comment by robmarkcole** [comment]
Can I just check:

1. You have double checked your api keys and they are in secrets.yaml?
2. The config you posted is correctly under the `sensors` heading?

Btw please use the 'insert code' formatting when pasting, makes if much easier to read.
Cheers

**Comment by robmarkcole** [comment]
OK solved the error in 5cea6b219d56ae51f3166f7ed78f071d8bfe48ef 
Most likely you have either entered invalid credentials or have somehow burned through the 1000 requests limit. Either way the sensor will show the error as the state of the sensor
